actionpack/lib/action_dispatch/journey/router.rb

```diff
@@ -122,6 +122,7 @@ def find_routes req
 
         def match_head_routes(routes, req)
           head_routes = match_routes(routes, req)
+          head_routes.delete_if { |route| route.verb == // }
```

Comment:
@tgxworld I went through this with @tsun1215 at the Open Academy yesterday and I think it's probably the least invasive method of doing it at the moment that we can backport to 4-2-stable. I suspect that this block of code will be rewritten again before Rails 5.0 ships